go/netutil/conn.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netutil
+
+import (
+	"net"
+	"time"
+)
+
+var _ net.Conn = (*connWithTimeout)(nil)
+
+type connWithTimeout struct {
+	conn         net.Conn
+	readTimeout  time.Duration
+	writeTimeout time.Duration
+}
```

Comment:
I think a better way would be this:
```
type connWithTimeout struct {
  net.Comm
  ...
}
```

Then you only have to define Read and Write.